### PR TITLE
Allow testing CI locally with act

### PR DESCRIPTION
**Pull request by i386x**
Adjust `.github/workflows/ci.yaml` and related files so it can be tested locally using `act` tool and describe the process in `DEVELOPING.md`.

### CODE REVIEW

**Review by ctron — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -31,6 +31,10 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
```
[the remark this answers is not in the document]
Comment (i386x) [reply .github/workflows/ci.yaml]: @ctron thanks for the note, I made a local action for that. Just to be clear this PR should have no effect on what is running inside GitHub CI (due to the `if: ${{ env.ACT }}` conditions). Also `act` does not use original GitHub-maintained Ubuntu images but Ubuntu images maintained by `act` upstream (they are very similar but not the same).

**Review by jcrossley3 — CHANGES_REQUESTED** [review]
What problem are we solving with act?

File: .github/workflows/ci.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ctron) [reply .github/workflows/ci.yaml]: >  Also act does not use original GitHub-maintained Ubuntu images but Ubuntu images maintained by act upstream (they are very similar but not the same).

That makes a bit uncomfortable. As it kind of voids the idea of running the same (the exact same) workflow locally.

It feels we're adding a lot of stuff, and I don't really see the benefit.

**Comment by i386x** [comment]
Since we are using `xtask` now, this change is no longer needed as everything we need to do (in context of project development/testing) can be managed by `cargo` and some `xtask` action.

**Pull request closed without merging**